include/libdnf/conf/config_main.hpp

```diff
@@ -76,6 +76,8 @@ class ConfigMain : public Config<Option::Priority::MAINCONFIG> {
     const OptionNumber<std::int32_t> & log_size() const;
     OptionNumber<std::int32_t> & log_rotate();
     const OptionNumber<std::int32_t> & log_rotate() const;
+    OptionPath & debugdir();
+    const OptionPath & debugdir() const;
```

Comment:
Does this need to be another config option? IMO it's unnecessarily growing the number of config options we have. The directory is normally a one-time affair, it should be potentially specifiable on the command line, but a permanent config value doesn't make much sense to me.

Reply:
Yes and no. Another point of view:
* Writing of the debug data - the `debug_solver` configuration option - can also be activated in the configuration file.  Therefore, if the user can activate the writing of debug data in the configuration file, it also makes sense to enable settings in the configuration file where to write debug data.
* All configuration options supported by configuration file, can be set from the commandline using `setopt=`. But if we have a special option for debug dir (not part of the configuration file) we will need a new command line option to set it. The number of command line arguments we have is growing. Or some hack for `setopt=`, because setopt supports the same set of options as supported in the configuraton file.

Reply:
> * Writing of the debug data - the `debug_solver` configuration option - can also be activated in the configuration file.

Well, that also doesn't make much sense to me.

> * All configuration options supported by configuration file, can be set from the commandline using `setopt=`. But if we have a special option for debug dir (not part of the configuration file) we will need a new command line option to set it. The number of command line arguments we have is growing.

A new CLI option seems like a good way to do it here and to an extent a growing number of CLI options is unavoidable - dnf has a lot of features. Bundling it under setopt is a cheap way out but it's a poor user experience - the setopt options aren't listed by `--help` and they are more cumbersome to use. IMO using `--setopt` should be an exception for when you need to override a config option, but it shouldn't be used instead of a proper CLI argument.